ocean_lib/models/datatoken_base.py

```diff
@@ -394,9 +422,26 @@ def build_access_service(
             consumer_parameters=consumer_parameters,
         )
 
+    @enforce_types
     def get_publish_market_order_fees(self):
         return TokenFeeInfo.from_tuple(self.contract.getPublishingMarketFee())
 
+    @enforce_types
+    def approve_publish_market_order_fees(self, tx_dict: dict):
+        fees = self.get_publish_market_order_fees()
+        if fees.amount > 0 and fees.address != ZERO_ADDRESS:
```

Comment:
Recommend: change this line to:
```python
if fees.amount == 0 or fees.address == ZERO_ADDRESS or fees.token == ZERO_ADDRESS:
    return
```

And also un-indent the rest of the method.

Why: 
- `fees.token` portion added, because it's also a reason to skip. And, `ERC20Template3.sol::orderStarted()` has similar logic. 
- simpler & easier to maintain: you avoid having to indent the remaining 11 lines of this method.